libs/providers/flipt-web/src/lib/flipt-web-provider.ts

```diff
@@ -0,0 +1,263 @@
+import {
+  EvaluationContext,
+  Provider,
+  JsonValue,
+  ResolutionDetails,
+  Logger,
+  ProviderStatus,
+  StandardResolutionReasons,
+  ErrorCode,
+  TypeMismatchError,
+  GeneralError,
+} from '@openfeature/web-sdk';
+import { FliptEvaluationClient } from '@flipt-io/flipt-client-browser';
+import { EvaluationReason, FliptWebProviderOptions } from './models';
+import { transformContext } from './context-transformer';
+
+export class FliptWebProvider implements Provider {
+  metadata = {
+    name: FliptWebProvider.name,
+  };
+
+  // logger is the Open Feature logger to use
+  private _logger?: Logger;
+
+  // status of the provider
+  private _status: ProviderStatus = ProviderStatus.NOT_READY;
+
+  // namespace is the namespace to use for the evaluation
+  private _namespace: string;
+
+  // options is the options provided to the provider
+  private _options?: FliptWebProviderOptions;
+
+  // client is the Flipt client reference
+  private _client?: FliptEvaluationClient;
+
+  // globalContext is the context that will be merged with all the evaluations if provided
+  private _globalContext: EvaluationContext = {};
+
+  readonly runsOn = 'client';
+
+  hooks = [];
+
+  constructor(namespace = 'default', options?: FliptWebProviderOptions, logger?: Logger) {
+    this._namespace = namespace;
+    this._options = options;
+    this._logger = logger;
+  }
+
+  get status() {
+    return this._status;
+  }
+
+  async initialize(context?: EvaluationContext | undefined): Promise<void> {
+    return Promise.all([this.initializeClient()])
+      .then(() => {
+        this._globalContext = context || {};
+        this._status = ProviderStatus.READY;
+        this._logger?.info('FliptWebProvider initialized');
+      })
+      .catch((error) => {
+        this._status = ProviderStatus.ERROR;
+        this._logger?.error('FliptWebProvider initialization failed', error);
```

Comment:
The SDK now manages state automatically. The provider is considered `ready` if the `initialize` method runs successfully. Please either remove this catch block or rethrow.